### PR TITLE
New Addon: Show message count in page title

### PR DESCRIPTION
**Pull request by 8bitjake**
**Resolves**

No issues are resolved. This is just a nice addon to have (in my opinion).

**Changes**

Added a new addon, the Message count in Title addon. It shows your current message count in the Scratch webpage, along with showing a new page icon when new messages are received.

**Reason for changes**

Again, it's a quality-of-life improvement addon. Maybe.

**Tests**

I test all my changes by adding them to my experimental download of SA. It's now my main version, so I always test my addons in a very easy way. And, yes, this was tested.

### CODE REVIEW

**Comment by FunctionalMetatable** [comment]
I think this addon should be merged with msg-count-badge

File: addons/msg-title/addon.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Message count in page title",
+  "description": "Show your message count in the Scratch page's title.",
+  "credits": [
+    {
```
Comment (RedGuy12) [inline addons/msg-title/addon.json:5]: I kinda would like credit, seeing as I made most of it...

File: addons/msg-title/addon.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      "name": "8bitjake",
+      "link": "https://scratch.mit.edu/users/8bitjake"
+    }
+  ],
+  "userscripts": [
+    {
+      "url": "background.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "runAtComplete": true
```
Comment (RedGuy12) [inline addons/msg-title/addon.json:14]: `true` is the default value for `runAtComplete`. So this is unneeded.

File: addons/msg-title/addon.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    {
+      "url": "favicon.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "settingsMatch": { "id": "msgIcon", "value": true },
+      "runAtComplete": true
+    }
+  ],
+  "tags": ["community", "recommended"],
```
Comment (RedGuy12) [inline addons/msg-title/addon.json:23]: Not recommended unless W_L says so.

File: addons/msg-title/background.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default async function ({ addon, msg, console }) {
+  var oldTitle = document.title;
```
Comment (RedGuy12) [inline addons/msg-title/background.js:2]: `  if (!addon.auth.isLoggedIn) return;` to fix that bug

Comment (RedGuy12) [reply addons/msg-title/background.js:2]: Also please resolve the conflicts

**Comment by RedGuy12** [comment]
Also please resolve the conflicts

File: addons/msg-title/addon.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "url": "background.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "runAtComplete": true
+    },
+    {
+      "url": "favicon.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "settingsMatch": { "id": "msgIcon", "value": true },
+      "runAtComplete": true
```
Comment (RedGuy12) [inline addons/msg-title/addon.json:20]: You removed the settings, remove this too

File: addons/msg-title/addon.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "userscripts": [
+    {
+      "url": "background.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "runAtComplete": true
+    },
+    {
+      "url": "favicon.js",
+      "matches": ["https://scratch.mit.edu/*"],
+      "settingsMatch": { "id": "msgIcon", "value": true },
```
Comment (RedGuy12) [inline addons/msg-title/addon.json:19]: You removed the settings, so remove this too.

**Comment by WorldLanguages** [comment]
TBH I'm not entirely sure whether this addon in its current from is a good idea. Specific points:
1. It's redundant with the `msg-count-badge` addon in a way
2. Chrome now [throttles JS timers](https://developer.chrome.com/blog/timer-throttling-in-chrome-88/), so it would be a better idea to update the tab titles from a persistent script.

**Comment by RedGuy12** [comment]
Found a bug: project titles are loaded late so the oldTitle variable is blank on page load

File: addons/msg-title/favicon.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
...JDiGZ2yhNdKKmZzM4yOKj4t/jF91r/QEjqvTb5Ho16HHRU4K677X2Bbsc0FlKgGw6a1cnSpQ+R7cnImF5qR4Z8M+998cramyi2rMMFiSTyAYR0crSzDR6a1c/mqM8q9AFNghAXkTQpUn7HO5aaS3SFaMRoMkytNgVitG68fHnzNYkk8gS0JBh8NGKdAz4DBxECdBdlwtOQXm4uiWs3SbRtIVhEqbLLLqSYbpqlEwjkCylHPdvgo0LOBjoiB5ha+41IaqUOB1GLhFnOep4y3ienBGvGmB35pia/XOGoNza4H8RHOtnRMD5CeVL6gMS68lMhEPcZobrhlPDjnhB/4jOABHj8sYkE8gVHu7HB/VTwEdgvamXCLyI0aqkg1h0h30e71khY+an4ox60iVLX6xrVQNkZLRMJ5AlrHO3GBvdjgT7cbJDZqbVIeMFUlrFFWjBqfiBXsGAyd37ay+K7KFUvVH8HUy84IYlAvlDhaDc2aB9JDukOquqFsATVbf9gdg9b7WrGNtMLQer0rBLNhoM+H1EBZjR0Cl5BmX52EoG8gJB14O0iFQL/zgiZy9GuR3Q97qUkWDGcKvePlrwSNiWI1lWutjIgv+h6w5X3tUV3JJLIBxByiaPd2KB9BNDn5+rw+54VMjhRaoX4SwpIdo+5xRECdGTqpxOfS34r/vyP8Z0pEhOOdXdyzoPtEok0E8k5JGNQ1qSobyB05LjOiHBmduN1/3dRP/HxF183s0tYUJVQkoyiVB//tH1qu3OV7Hu8iYulUklEmgMtGg88Gg0NERcCfQCatf9rphlrc12/h5WFprYyj50RF66fZjaydUW4v20cPjyJUJOAiOTSJMkQcCK4ExTiL7oPm2X6EkC6psq8AGBeBKheb3ZRNKccyFb06BEmEfuG6Oko5mb0q1NMxTyCEH2+vgEcmGWCcb3mIn0BYCHQN4XGLX311RkJ5HKBTqfcR2J/g4Bm+rsg0RdQj+Rm/Keaw2SOuj+mu5Z38lmDyOUV7dv/urhVq4IEsgfCpJx94kBG5/oijTr/9RyfB5+AaWANwmqAbqQpzDtYYA4YAnQn5l7+dy44hXN9H+skjq20TW1byeldVqL7EU/pd/zWrdkCkgwS+taOPm54CwFTwSbOPaDvX20Fc8F/+awPUc/LED681qLF/wBb6icWdpxK1gAAAABJRU5ErkJggg==`;
```
Comment (GrahamSH-LLK) [inline addons/msg-title/favicon.js:8]: This icon is really blurry

Comment (RedGuy12) [reply addons/msg-title/favicon.js:8]: it's the same one scratch normally uses, except with a red dot on it

Comment (GrahamSH-LLK) [reply addons/msg-title/favicon.js:8]: Wait, you coded this?

Comment (RedGuy12) [reply addons/msg-title/favicon.js:8]: basically yeah

on the discord he just asked "what next?", I figured it out, and he added it to his branch

**Pull request closed without merging**